frontend/public/components/monitoring/alert-manager-utils.tsx

```diff
@@ -36,6 +36,15 @@ export const patchAlertManagerConfig = (
   secret: K8sResourceKind,
   yaml: object | string,
 ): Promise<any> => {
+  if (_.isString(yaml)) {
+    try {
+      safeLoad(yaml);
+    } catch (e) {
+      return new Promise(() => {
+        throw new Error(`Error parsing Alertmanager YAML: ${e}`);
```

Comment:
Use `Promise.reject`, but I don't think it's necessary if you just make the check inside of `save()`.